examples/turbulent_advection-diffusion.jl

```diff
@@ -0,0 +1,181 @@
+MultiLayerQG.set_q!(MQGprob, q₀)
+nothing 
+
+# ## Tracer advection-diffusion setup
+#
+# Now that we have a `MultiLayerQG.Problem` setup to generate our turbulent flow, we
+# setup an advection-diffusion simulation. This is done by passing the `MultiLayerQG.Problem`
+# as an argument to `TracerAdvectionDiffusion.Problem` which sets up an advection-diffusion problem
+# with same parameters where applicable. We also need to pass a value for the constant diffusivity `κ`,
+# the `stepper` used to step the problem forward and when we want the tracer released into the flow.
+# We will let the flow run until it reaches a statistical equilibrium and then advect-diffuse the tracer.
+
+κ = 0.002
+nsteps = 4000                    # total number of time-steps
+nsubs = 1                        # number of steps the simulation takes at each iteration 
+tracer_release_time = dt * 8000  # run flow for some time before releasing tracer
+
+ADprob = TracerAdvectionDiffusion.Problem(dev, MQGprob; κ, stepper, tracer_release_time)
+nothing
+
+# ## Initial condition for concentration in both layers
+#
+# We have a two layer system so we will advect-diffuse the tracer in both layers.
+# To do this we set the initial condition for tracer concetration as a Gaussian centered at the origin.
+# Then we create some shortcuts for the `TracerAdvectionDiffusion.Problem`.
+gaussian(x, y, σ) = exp(-(x^2 + y^2) / (2σ^2))
+
+amplitude, spread = 10, 0.15
+c₀ = [amplitude * gaussian(x[i], y[j], spread) for j=1:grid.ny, i=1:grid.nx]
+
+TracerAdvectionDiffusion.set_c!(ADprob, c₀)
+
+# Shortcuts for advection-diffusion problem
+sol, clock, vars, params, grid = ADprob.sol, ADprob.clock, ADprob.vars, ADprob.params, ADprob.grid
+x, y = grid.x, grid.y
+
+# ## Saving output
+#
+# The parent package `FourierFlows.jl` provides the functionality to save the output from our simulation.
+# To do this we write a function `get_concentration` and pass this to the `Output` function along 
+# with the `TracerAdvectionDiffusion.Problem` and the name of the output file.
+
+function get_concentration(prob)
+  MultiLayerQG.invtransform!(prob.vars.c, deepcopy(prob.sol), prob.params.MQGprob.params)
+
+  return prob.vars.c
+end
+
+output = Output(ADprob, "advection-diffusion.jld2", (:concentration, get_concentration))
+
+# This saves information that we will use for plotting later on
+saveproblem(output)
+
+# ## Step the problem forward and save the output
+#
+# We specify that we would like to save the concentration every 50 timesteps using `save_frequency`
+# then step the problem forward.
+save_frequency = 50 # Frequency at which output is saved
+
+startwalltime = time()
+while clock.step <= nsteps
+  if clock.step % save_frequency == 0
+    saveoutput(output)
+    log = @sprintf("Output saved, step: %04d, t: %.2f, walltime: %.2f min",
+                   clock.step, clock.t, (time()-startwalltime) / 60)
+
+    println(log)
+  end
+
+  stepforward!(ADprob)
+  TracerAdvectionDiffusion.updatevars!(ADprob)
```

Comment:
is this `TracerAdvectionDiffusion.updatevars!(ADprob)` needed here?

Reply:
With the new `get_concentration` function I think you are right this is not needed.